vignettes/netCDF_Functionality.Rmd

```diff
@@ -0,0 +1,599 @@
+the array should have 10 columns, and the "name" and "unit" field ashould each
+have 10 values.
+
+Use CMIP6 standard variable names where available.The [day](https://github.com/PCMDI/cmip6-cmor-tables/blob/master/Tables/CMIP6_day.json)
+and [Eday](https://github.com/PCMDI/cmip6-cmor-tables/blob/master/Tables/CMIP6_Eday.json)
+tables for time-varying variables and [fx](https://github.com/PCMDI/cmip6-cmor-tables/blob/master/Tables/CMIP6_fx.json)
+and [Efx](https://github.com/PCMDI/cmip6-cmor-tables/blob/master/Tables/CMIP6_Efx.json)
+for time-invariant variables.
+
+
+```{r, VarAtt, results = 'hide'}
+# single variable
+var_attributes <- list(
+  name = 'WatYrWDD',
+  long_name = 'Water Year Wet Degree Days',
+  units = 'number of days'
+)
+
+# multiple variables
+var_attributes2 <- list(
+    name = paste0('VarDD_', 1:10),
+    units = rep('number of days', 10)
+)
+
+```
+
+### Time Attributes
+
+A named list describing the time dimension. The *name*, *units*, *calender*, and *vals*
+fields are required here.
+
+The length "vals" in the time_attribute list needs to be the same values as the
+number of columns in the array.
+
+A common unit for netCDF is days since 1900. A full list of accepted time units can
+be found [here](https://ncics.org/portfolio/other-resources/udunits2/).
+
+```{r, TimeAtt, results = 'hide'}
+time_attributes <- list(
+  name = 'time',
+  units = 'days since 1900-01-01',
+  calendar = 'standard',
+  unlim = 'TRUE',
+  vals = time_midpoints # mid point of year
+  )
+```
+
+### Vertical (depth / Z) Attributes
+
+A named list describing the vertical (depth / Z) dimension. The *name*, *units*,
+and *vals* fields are required.
+
+The length of the "vals" in the vertical_attribute list need to match the information
+in the array passed (either the number of cols if there is no time dimension, or the length
+of the third dimension if there is a time dimension).
+
+```{r, VertAtt, results = 'hide'}
+vertical_attributes <- list(
+  name = 'depth',
+  units = 'cm',
+  positive = 'down', # depth is down, height is up
+  vals = depths_max
+)
+```
+
+### CRS Attributes
+
+The CRS attributes describe the coordinate reference system (CRS).
+
+The only required field in the CRS attribute list is the 'proj' field. The proj
+field is the the CRS is proj4string from the PROJ.4 library.
+
+Additionally, we strongly recommend defining the CRS in the crs_wkt format, which
+is preferred by [netCDF convention] (http://cfconventions.org/cf-conventions/cf-conventions.html#use-of-the-crs-well-known-text-format).
+
+The *wkt* function in the raster package can be used to generate wkts.
+
+```{r, CRSAtt, results = 'hide'}
+
+crs_attributes <- list(
+ proj = '+init=epsg:4326',
+  grid_mapping_name = "latitude_longitude",
+  longitude_of_prime_meridian = 0.0,
+  semi_major_axis = 6378137.0,
+  inverse_flattening = 298.257223563
+)
+
+locationsSP <- sp::SpatialPoints(locations)
+sp::proj4string(locationsSP) <- sp::CRS(crs_attributes$proj)
```

Comment:
why do this in two lines 260-261 instead of one:
`locationsSP <- sp::SpatialPoints(coords = locations, proj4string = sp::CRS(crs_attributes$proj))